### PR TITLE
use user.Current() for getting current user

### PR DESCRIPTION
**Pull request by mattn**
on windows, whoami getting error: `cannot find name for user ID -1`

### CODE REVIEW

**Comment by EricLagerg** [comment]
Ugh. Apparently os.Geteuid is a noop on windows: https://golang.org/src/syscall/syscall_windows.go#L964

File: whoami/whoami.go [hunk 1]
```diff
@@ -73,10 +72,9 @@ func main() {
 		os.Exit(0)
 	}
 
-	uid := strconv.Itoa(os.Geteuid())
-	u, err := user.LookupId(uid)
+	u, err := user.Current()
 	if err != nil {
-		fatal.Fatalf("cannot find name for user ID %s\n", uid)
+		fatal.Fatalln("cannot find name for current user")
```
Comment (EricLagerg) [inline whoami/whoami.go]: I'm not a huge fan of having the outputs be different... but I couldn't find a way around it.

**Comment by EricLagerg** [comment]
Before I merge this, could you move the windows version to whoami_windows.go and move the original to whoami_unix.go?

**Comment by mattn** [comment]
separated sources for unix/windows.

**Comment by EricLagerg** [comment]
awesome, thank you mattn!

**Pull request merged**